modules/nlu-testing/src/tools/visualisation.ts

```diff
@@ -0,0 +1,189 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+import { distance, similarity } from 'ml-distance'
+import { Matrix } from 'ml-matrix'
+import { PCA } from 'ml-pca'
+
+import { BotState } from '../backend/typings'
+const clustering = require('density-clustering')
+
+export async function computeEmbeddingSimilarity(state: BotState) {
+  const intentDatas = _.groupBy(state.trainDatas, 'intent')
+
+  const intentEmb = {}
+  for (const key in intentDatas) {
+    if (Object.prototype.hasOwnProperty.call(intentDatas, key)) {
+      const meanEmb = intentDatas[key]
+        .map(o => o.utt_emb)
+        .reduce((tot, cur) => {
+          return Matrix.add(tot, new Matrix([cur]))
+        }, Matrix.zeros(1, intentDatas[key][0].utt_emb.length))
+      intentEmb[key] = Matrix.div(meanEmb, intentDatas[key].length).to1DArray()
```

Comment:
Please add spaces to make the code more readable. This applies to the whole PR, there's multiple big blocks and it makes it harder to see what happens